crates/eframe/src/native/file_storage.rs

```diff
@@ -10,13 +10,80 @@ use std::{
 /// [`egui::ViewportBuilder::app_id`] of [`crate::NativeOptions::viewport`]
 /// or the title argument to [`crate::run_native`].
 ///
-/// On native the path is picked using [`directories::ProjectDirs::data_dir`](https://docs.rs/directories/5.0.1/directories/struct.ProjectDirs.html#method.data_dir) which is:
+/// On native, the path is:
 /// * Linux:   `/home/UserName/.local/share/APP_ID`
 /// * macOS:   `/Users/UserName/Library/Application Support/APP_ID`
-/// * Windows: `C:\Users\UserName\AppData\Roaming\APP_ID`
+/// * Windows: `C:\Users\UserName\AppData\Roaming\APP_ID\data`
 pub fn storage_dir(app_id: &str) -> Option<PathBuf> {
-    directories::ProjectDirs::from("", "", app_id)
-        .map(|proj_dirs| proj_dirs.data_dir().to_path_buf())
+    use egui::os::OperatingSystem as OS;
+    use std::env::var_os;
+    match OS::from_target_os() {
+        OS::Nix => var_os("XDG_DATA_HOME")
+            .map(PathBuf::from)
+            .filter(|p| p.is_absolute())
+            .or_else(|| home::home_dir().map(|p| p.join(".local").join("share")))
+            .map(|p| {
+                p.join(
+                    app_id
+                        .to_lowercase()
+                        .replace(|c: char| c.is_ascii_whitespace(), ""),
+                )
+            }),
+        OS::Mac => home::home_dir().map(|p| {
+            p.join("Library")
+                .join("Application Support")
+                .join(app_id.replace(|c: char| c.is_ascii_whitespace(), "-"))
+        }),
+        OS::Windows => roaming_appdata().map(|p| p.join(app_id).join("data")),
+        OS::Unknown | OS::Android | OS::IOS => None,
+    }
+}
+
+// Adapted from
+// https://github.com/rust-lang/cargo/blob/6e11c77384989726bb4f412a0e23b59c27222c34/crates/home/src/windows.rs#L19-L37
+#[cfg(all(windows, not(target_vendor = "uwp")))]
+#[allow(unsafe_code)]
+fn roaming_appdata() -> Option<PathBuf> {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStringExt;
+    use std::ptr;
+    use std::slice;
+
+    use windows_sys::Win32::Foundation::S_OK;
+    use windows_sys::Win32::System::Com::CoTaskMemFree;
+    use windows_sys::Win32::UI::Shell::{
+        FOLDERID_RoamingAppData, SHGetKnownFolderPath, KF_FLAG_DONT_VERIFY,
+    };
+
+    extern "C" {
+        fn wcslen(buf: *const u16) -> usize;
+    }
+    unsafe {
+        let mut path = ptr::null_mut();
+        match SHGetKnownFolderPath(
```

Comment:
Could also use the [known-folders](https://crates.io/crates/known-folders) crate, which has this function and nothing else. But it's still another dependency, which might end up pulling a different version of windows-sys, so I don't know if it's worth it.